internal/component/azure/servicebus/subscription.go

```diff
@@ -289,12 +324,65 @@ func (s *Subscription) ReceiveAndBlock(handler HandlerFunc, lockRenewalInSec int
 // Close the receiver and stops watching for new messages.
 func (s *Subscription) Close(closeCtx context.Context) {
 	s.logger.Debugf("Closing subscription to %s", s.entity)
+	s.cancel()
+}
 
-	// Ensure subscription entity is closed.
-	if err := s.receiver.Close(closeCtx); err != nil {
-		s.logger.Warnf("Error closing subscription for %s: %+v", s.entity, err)
+type LockRenewalOptions struct {
+	RenewalInSec int
+	TimeoutInSec int
+}
+
+func (s *Subscription) RenewLocksBlocking(ctx context.Context, receiver Receiver, opts LockRenewalOptions) error {
+	if receiver == nil {
+		return nil
+	}
+
+	shouldRenewLocks := opts.RenewalInSec > 0
+	if !shouldRenewLocks {
+		s.logger.Debugf("Lock renewal for %s disabled", s.entity)
+		return nil
+	}
+
+	t := time.NewTicker(time.Second * time.Duration(opts.RenewalInSec))
+	defer t.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			s.logger.Infof("context canceled while renewing locks for %s", s.entity)
+			return nil
+		case <-t.C:
+			// Check if the context is still valid
+			if ctx.Err() != nil {
+				s.logger.Infof("context canceled while renewing locks for %s", s.entity)
+				return nil
+			}
+			if s.requireSessions {
+				sessionReceiver := receiver.(*SessionReceiver)
+				if err := sessionReceiver.RenewSessionLocks(ctx, opts.TimeoutInSec); err != nil {
+					s.logger.Errorf("error renewing session locks for %s: %s", s.entity, err)
```

Comment:
Should this be a Warn log? Because locks could fail to be renewed if, for example, a message was ACK'd in the meantime. Not all errors here are necessarily bad.

Reply:
Yes definitely that's a mistake, fixed, ty.